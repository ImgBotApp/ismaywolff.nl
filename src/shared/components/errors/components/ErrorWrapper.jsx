/* eslint-disable default-case, consistent-return */

import { oneOf } from 'prop-types'
import styled from 'styled-components'

const setBackground = props => {
  switch (props.type) {
    case 'info':
      return 'var(--color-blue-light)'
    case 'error':
      return 'var(--color-red-light)'
  }
}

const setColor = props => {
  switch (props.type) {
    case 'info':
      return 'var(--color-blue-dark)'
    case 'error':
      return 'var(--color-red-dark)'
  }
}

const ErrorWrapper = styled.div`
  background: ${setBackground};
  color: ${setColor};
  fill: ${setColor};
  padding: var(--size-small) var(--size-regular);
`

ErrorWrapper.propTypes = {
  type: oneOf(['error', 'info']).isRequired
}

export default ErrorWrapper
