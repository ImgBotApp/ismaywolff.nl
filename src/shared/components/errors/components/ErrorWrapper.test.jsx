/**
 * @jest-environment node
 */

import 'jest-styled-components'
import React from 'react'
import { shallow } from 'enzyme'
import ErrorWrapper from './ErrorWrapper'

describe('<ErrorWrapper />', () => {
  it('accepts an error type', () => {
    const wrapper = shallow(<ErrorWrapper type="error" />)
    expect(wrapper).toMatchStyledComponentsSnapshot()
  })

  it('accepts an info type', () => {
    const wrapper = shallow(<ErrorWrapper type="info" />)
    expect(wrapper).toMatchStyledComponentsSnapshot()
  })
})
