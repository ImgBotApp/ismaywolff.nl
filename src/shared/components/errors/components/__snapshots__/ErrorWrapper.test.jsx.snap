// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorWrapper /> accepts an error type 1`] = `
.c0 {
  background: var(--color-red-light);
  color: var(--color-red-dark);
  fill: var(--color-red-dark);
  padding: var(--size-small) var(--size-regular);
}

<div
  className="c0"
  type="error"
/>
`;

exports[`<ErrorWrapper /> accepts an info type 1`] = `
.c0 {
  background: var(--color-blue-light);
  color: var(--color-blue-dark);
  fill: var(--color-blue-dark);
  padding: var(--size-small) var(--size-regular);
}

<div
  className="c0"
  type="info"
/>
`;
